docs/admin/monitoring.rst

```diff
@@ -173,17 +173,45 @@ The ``NodeInfo`` JMX MBean exposes information about the current node;
+| Name                    | Description                                        |
++=========================+====================================================+
+| ``NodeId``              | Provides the unique identifier of the node in the  |
+|                         | cluster.                                           |
++-------------------------+----------------------------------------------------+
+| ``NodeName``            | Provides the human friendly name of the node.      |
++-------------------------+----------------------------------------------------+
+| ``ClusterStateVersion`` | Provides the version of the current applied        |
+|                         | cluster state.                                     |
++-------------------------+----------------------------------------------------+
+| ``Role``                | The role of the node, possible roles are: MASTER   |
+|                         | DATA.                                              |
++-------------------------+----------------------------------------------------+
+| ``Primaries``           | The number of primary shards located on the node.  |
++-------------------------+----------------------------------------------------+
+| ``Replicas``            | The number of replica shards located on the node.  |
++-------------------------+----------------------------------------------------+
+| ``Unassigned``          | The number of unassigned shards. If the role of    |
+|                         | the node is MASTER this will show the total        |
+|                         | number of unassigned shards in the cluster,        |
+|                         | otherwise 0.                                       |
++-------------------------+----------------------------------------------------+
+| ``Shards``              | Information about the shards located on the node.  |
++-------------------------+----------------------------------------------------+
+
+``Shards`` returns a set of `CompositeData`_ objects containing detailed
+information of each shard located on the node with the following attributes:
 
-   Provides the version of the current applied cluster state
-
- - ``NodeId``
-
-   Provides the unique identifier of the node in the cluster
-
- - ``NodeName``
-
-   Provides the human friendly name of the node
++------------------+------------------------------------------------------+
+| Name             | Description                                          |
++==================+======================================================+
+| ``id``           | The id of the shard.                                 |
++------------------+------------------------------------------------------+
+| ``size``         | The size in bytes of the shard.                      |
++------------------+------------------------------------------------------+
+| ``state``        | The state of the shard, possible states are:         |
```

Comment:
You mentioned there should be an option to filter by shard state ?

Reply:
Well, it's more about thinking about how users are going to consume these metrics. We'll need to extend https://github.com/crate/jmx_exporter to also expose these metrics for prometheus and need to come up with a format that is useful for users who want to consume them.